decode.go

```diff
@@ -253,6 +253,16 @@ func (de *decoder) putvalue(wiretype int, val reflect.Value,
 
 	// Embedded message
 	case reflect.Struct:
+		bm := reflect.TypeOf((*encoding.BinaryUnmarshaler)(nil)).Elem()
```

Comment:
I have no clue about that. For my very basic understanding, it will put this variable on the stack of the function and create it each time you run into that part, but I'm sure Im missing some compiler optimizations here...
Is it not possible to `const bm = reflect.TypeOf((*encoding.BinaryUnmarshaler)(nil)).Elem()` since everything is static ?